packages/jsii-diff/lib/classes-ifaces.ts

```diff
@@ -0,0 +1,203 @@
+import reflect = require('jsii-reflect');
+import log4js = require('log4js');
+import { Analysis, FailedAnalysis, isSuperType } from './type-analysis';
+import { ComparisonContext, shouldInspect } from './types';
+
+const LOG = log4js.getLogger('jsii-diff');
+
+/**
+ * Compare two class types
+ *
+ * We require that all stable properties and methods on the original are
+ * present on the new type, and that they match in turn.
+ */
+export function compareClass(original: reflect.ClassType, updated: reflect.ClassType, context: ComparisonContext) {
+  if (updated.abstract && !original.abstract) {
+    context.mismatches.report(original, 'has gone from non-abstract to abstract');
+  }
+
+  for (const [origMethod, updatedMethod] of memberPairs(original, original.methods, updated, context)) {
+    compareMethod(original, origMethod, updatedMethod, context);
```

Comment:
Will this take the constructor into account? There is a nuance with ctors: if they’re are not declared, the base class ctor is used  (I think we mightn’t Be ironing this out in the jsii complier,so it might not be an issue)

Reply:
It will include the contructor, but you're right that I overlooked method inheritance.

Reply:
I was overlooking the constructor as well. I was under the impression it would be in the methods array but it was not. :/